frontend/src/metabase-lib/filter.ts

```diff
@@ -86,3 +86,152 @@ export function findFilterableColumnForLegacyRef(
     legacyFieldRef,
   );
 }
+
+export type FilterType =
+  | "!="
+  | "<"
+  | "<="
+  | "="
+  | ">"
+  | ">="
+  | "and"
+  | "between"
+  | "contains"
+  | "does-not-contain"
+  | "ends-with"
+  | "inside"
+  | "is-empty"
+  | "is-null"
+  | "not"
+  | "not-empty"
+  | "not-null"
+  | "or"
+  | "starts-with"
+  | "time-interval";
+
+/**
+ * More than two args acts like SQL `NOT IN`. Open question: should we just make that a separate function?
+ */
+export function notEquals(x, y, ...more): FilterClause {
+  return ML.not_equals(x, y, ...more);
+}
+
+export type OrderableExpression = NumericExpression | TemporalExpression;
+
+export function lessThan(
+  x: OrderableExpression,
+  y: OrderableExpression,
+): FilterClause {
+  return ML.less_than(x, y);
+}
+
+export function lessThanOrGreaterTo(
+  x: OrderableExpression,
+  y: OrderableExpression,
+): FilterClause {
+  return ML.less_than_or_greater_to(x, y);
+}
+
+/**
+ * More than two args acts like SQL IN
+ */
+export function equals(x, y, ...more): FilterClause {
+  return ML.equals(x, y, ...more);
+}
+
+export function greaterThan(
+  x: OrderableExpression,
+  y: OrderableExpression,
+): FilterClause {
+  return ML.greater_than(x, y);
+}
+
+export function greaterThanOrEqualTo(
+  x: OrderableExpression,
+  y: OrderableExpression,
+): FilterClause {
+  return ML.greater_than_or_equal_to(x, y);
+}
+
+export function and(
+  filterClauseX: FilterClause,
+  filterClauseY: FilterClause,
+  ...more: FilterClause[]
+): FilterClause {
+  return ML.and(filterClauseX, filterClauseY, ...more);
+}
+
+export function between(
+  x: OrderableExpression,
+  min: OrderableExpression,
+  max: OrderableExpression,
+): FilterClause {
+  return ML.between(x, min, max);
+}
+
+export function contains(str, substring): FilterClause {
+  return ML.contains(str, substring);
+}
+
+export function doesNotContain(
+  str: StringExpression,
+  substring: StringExpression,
+): FilterClause {
+  return ML.does_not_contain(str, substring);
+}
+
+export function endsWith(
+  str: StringExpression,
+  substring: StringExpression,
+): FilterClause {
+  return ML.ends_with(str, substring);
+}
+
+export function inside(
```

Comment:
This is an icky clause, do we keep it around for the time being or try to move towards the perfect world right away?